### ISSUE TITLE
Weird 'token recognition error' error messages on stdout/stderr when evaluation fails

### ISSUE THREAD

**Issue by bitbonk** [sklose/NCalc2#118]
We have updgraded _CoreCLR-NCalc_ from version `3.1.251` to `3.1.253`. Now when an evaluation fails with an exception (as expected) there is also always some "weird" output on stdout (or stderr?) like this:

```log
line 1:5 token recognition error at: '$'
line 1:8 token recognition error at: '@'
line 1:9 token recognition error at: '#*%'
```

This can be reproduced by running the following xunit test on the console using `dotnet test`

```csharp
using FluentAssertions;
using NCalc;

namespace NCalcErrors;

public class UnitTest1
{
    [Theory]
    // ...
    [InlineData("+= x%$*%@#*%")]
    public void Evaluate_InvalidExpression_Throws(string expression)
    {
        var sut = new Expression(
            expression,
            EvaluateOptions.IgnoreCase |
            EvaluateOptions.MatchStringsWithIgnoreCase |
            EvaluateOptions.UseDoubleForAbsFunction);

        sut.Invoking(s => s.Evaluate()).Should().Throw<Exception>();
    }
}
```

**Comment by bitbonk** [comment]
Just discovered that behavior also existed in CoreCLR-NCalc version `3.1.251`.  It just that the upgrade from of the .NET SDK from version `8.0.302` to `8.0.400` made this visible. Running the above test with CoreCLR-NCalc version `3.1.251` and .NET SDK `8.0.302` does not result in the output but running the above test with CoreCLR-NCalc version `3.1.251`  and .NET DSK `8.0.400` suddenly does.

**Comment by david-brink-talogy** [comment]
The output is coming from the default error listener in ANTLR4.  
https://github.com/antlr/antlr4/blob/811b7fda58bd14d7f0abc496b6fd651dfa01ed97/runtime/CSharp/src/Lexer.cs#L559.

**Comment by bitbonk** [comment]
@david-brink-talogy Yes, it looks like the default error listener `ConsoleErrorListener.Instance` must be removed using `RemoveErrorListener()`
